### PR TITLE
Curtain systems 2016

### PR DESCRIPTION
**Pull request by moethu**
### Purpose

Curtain System By Face
### Declarations

Check these if you believe they are true
- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.
### Reviewers

@mjkkirschner 
### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/CurtainSystem.cs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        /// </summary>
+        private void InitCurtainSystem(Autodesk.Revit.DB.CurtainSystem curtainSystem)
+        {
+            InternalSetCurtainSystem(curtainSystem);
+        }
+
+        /// <summary>
+        /// Initialize a CurtainSystem element
+        /// </summary>
+        private void InitCurtainSystem(ReferenceArray faces, Autodesk.Revit.DB.CurtainSystemType curtainSystemType)
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/CurtainSystem.cs]: The logic in this function is not clear to me:
- How will element binding work?
- If there are multiple curtain systems created, which one will CurtainSystem be bound to?

File: src/Libraries/RevitNodes/Elements/CurtainSystem.cs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        /// <param name="curtainSystemType"></param>
+        /// <returns></returns>
+        public static CurtainSystem ByFaces(Autodesk.DesignScript.Geometry.Surface[] faces, CurtainSystemType curtainSystemType)
+        {
+            ReferenceArray ca = new ReferenceArray();
+
+            foreach (Autodesk.DesignScript.Geometry.Surface face in faces)
+            {
+                var reference = face.Tags.LookupTag("RevitFaceReference");
+                if (reference != null)
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/CurtainSystem.cs]: If it is null, a warning/error message will be better.

File: src/Libraries/RevitNodes/Elements/InternalUtilities/ElementWrapper.cs [hunk 3]
```diff
@@ -212,6 +212,11 @@ public static Mullion Wrap(Autodesk.Revit.DB.Mullion ele, bool isRevitOwned)
            return Mullion.FromExisting(ele, isRevitOwned);
         }
 
+        public static CurtainSystem Wrap(Autodesk.Revit.DB.CurtainSystem ele, bool isRevitOwned)
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/InternalUtilities/ElementWrapper.cs]: One method is required for CurtainSystemType as well.

File: src/Libraries/RevitNodes/Properties/Resources.resx [hunk 4]
```diff
@@ -312,6 +312,9 @@
   <data name="MultipleSurfacesIntroducedAfterConversion" xml:space="preserve">
     <value>There are multiple surfaces converted from the input face.</value>
   </data>
+  <data name="CurtainSystemTypeNotFound" xml:space="preserve">
+    <value>Curtain System Type not found.</value>
```
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Properties/Resources.resx]: The error message is better to be a complete sentence.

File: .gitignore [hunk 5]
```diff
@@ -219,3 +219,4 @@ test/Libraries/RevitIntegrationTests/RevitTestConfiguration.xml
 
 # Icon resources
 /src/DynamoRevitIcons/*.resources
+src/Libraries/RevitNodes/Elements/InternalUtilities/ElementWrapper.cs
```
Comment (moethu) [inline .gitignore]: certainly is, sorry.

File: src/Libraries/RevitNodes/Elements/CurtainSystem.cs (hunk 1 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/CurtainSystem.cs]: @Randy-Ma one Curtain System will / should be returned. In the Revit API you can only create the curtain system from an array of faces. The curtain system then has multiple references.

**Comment by moethu** [comment]
@Randy-Ma applied changes

File: src/Libraries/RevitNodes/Elements/CurtainSystem.cs (hunk 1 above)
Comment (Randy-Ma) [inline src/Libraries/RevitNodes/Elements/CurtainSystem.cs]: No, NewCurtainSystem2 can return multiple curtain systems.

**Comment by Randy-Ma** [comment]
@moethu 
Thanks! There are new comments added:
- Element binding behavior: NewCurtainSystem2 can return multiple curtain systems. This needs to be considered.
- A wrapping method for CurtainSystemType need to be implemented.

File: src/Libraries/RevitNodes/Elements/CurtainSystem.cs (hunk 1 above)
Comment (moethu) [inline src/Libraries/RevitNodes/Elements/CurtainSystem.cs]: Yes right, will change to accept only one surface and then bind to the first result of NewCurteinSystem2

**Comment by moethu** [comment]
@Randy-Ma accepting only one face now to solve the multiple faces/binding issue.
And added the type wrapper
```
        public static CurtainSystem Wrap(Autodesk.Revit.DB.CurtainSystem ele, bool isRevitOwned)
        {
            return CurtainSystem.FromExisting(ele, isRevitOwned);
        }

        public static CurtainSystemType Wrap(Autodesk.Revit.DB.CurtainSystemType ele, bool isRevitOwned)
        {
            return CurtainSystemType.FromExisting(ele, isRevitOwned);
        }
```

**Pull request closed without merging**